src/main/scala/chisel3/util/random/Lfsr.scala

```diff
@@ -0,0 +1,137 @@
+// See LICENSE for license details.
+
+package chisel3.util.random
+
+import chisel3._
+
+/** Linear Feedback Shift Register (LFSR) generator. This produces a maximal period Galois LFSR of the requested width.
+  * {{{
+  * // Instantiate a new 4-bit LFSR, with the default seed of 1, and automatically stepping.
+  * val lfsr4 = Module(new Lfsr(4))
+  * // Print out the LFSR values in order.
+  * val (count, _) = chisel3.util.Counter(true.B, 16)
+  * printf("  - %d: 0b%b\n", count, input)
+  * // The values this prints are below. The seed (0b0001) is first. The LFSR cycles after 2^4-1 steps.
+  * //  -  0: 0b0001
+  * //  -  1: 0b1100
+  * //  -  2: 0b0110
+  * //  -  3: 0b0011
+  * //  -  4: 0b1101
+  * //  -  5: 0b1010
+  * //  -  6: 0b0101
+  * //  -  7: 0b1110
+  * //  -  8: 0b0111
+  * //  -  9: 0b1111
+  * //  - 10: 0b1011
+  * //  - 11: 0b1001
+  * //  - 12: 0b1000
+  * //  - 13: 0b0100
+  * //  - 14: 0b0010
+  * //  - 15: 0b0001
+  * }}}
+  * @param width LFSR width
+  * @param seed an initial seed (this cannot be zero)
+  */
+class Lfsr(width: Int, seed: BigInt = 1) extends Module {
```

Comment:
Also minor nitpick, is `chisel3.util.random.Lfsr` preferred or `chisel3.util.random.LFSR`?

Reply:
I think this should be `LFSR`. I don't know if we have a formal convention on this, but most things that are acronyms are uppercase, e.g., `ValidIO`. For consistency this is changed in the rebase.